components/application/application.js

```diff
@@ -0,0 +1,204 @@
+// libraries
+import { body }  from 'express-validator';
+
+// validator
+import {  protectedRoute } from '../../middleware/protectedRoute.js';
+
+import validationGuard from '../../middleware/validationGuard.js'
+
+// schemas
+import Application from "../../schemas/application.js"
+
+// services
+import "../../services/emailer.js"
+
+
+
+
+async function postMakeApplication(req,res,ext){
+    // if multiple applications aren't allowed, we check if there's a single Application for user id where status is pending
+    // does this need to be a middleware? I don't think so
+
+    try{
+        if(process.env.ALLOW_MULTIPLE_APPLICATIONS !== "true"){
+            let application = await Application.findOne({
+                requestedBy  : req.auth.id,
+                status       : "pending"
+            }).exec()
+    
+            if(application !== null){
+                return res.status(400).json({
+                    err : "User has an application pending review."
+                })
+            }
+        }
+
+        let application = new Application({
+            amount       : parseFloat(req.body.amount),
+            payments     : parseInt  (req.body.payments),
+            paymentAmount: parseFloat(req.body.paymentAmount),
+            reason       : req.body.reason,
+            description  : req.body.description,
+            requestedBy  : req.auth.id,
+            status       : "pending"
+        })
+
+        // client made an invalid request
+        let err = application.validateSync()
+        if(err){
+            return res.status(400).json({
+                err : err.message
+            })
+        }
+
+        await application.save()
+
+        res.status(200).json({
+            msg  : "Application received.",
+        })
+    }catch(e){
+        return next(e)
+    }
+}
+
+
+async function getMyApplications(req,res,next){

```

Comment:
Would this be clearer?
```suggestion
async function getApplicationsForAuthenticatedUser(req,res,next){

```

Reply:
I guess so

Reply:
Done